src/tests/test_method_middleware.py

```diff
@@ -0,0 +1,324 @@
+from typing import Callable
+
+import pytest
+
+from polydatum import DataAccessLayer, Service
+from polydatum.dal import DataManager
+from polydatum.middleware import DalMethodRequest, DeferredDalAttrRequest, dal_method_resolver_middleware, \
+    handle_dal_method, dal_resolver
+from polydatum.errors import InvalidMiddleware
+
+
+class FooMiddleware:
+    # Does not implement __call__ and hence not
+    # a Callable
+    pass
+
+
+@pytest.mark.parametrize('middleware', [
+    (FooMiddleware, ),
+    (FooMiddleware(), ),
+    (True, ),
+    ('random-string',),
+    ([],),
+    (None,),
+])
+def test_dal_middleware_requires_callable(middleware):
+    """
+    Verify that you can specify method middleware as either an instance
+    or a class (as a convenience).
+    """
+    with pytest.raises(InvalidMiddleware):
+        DataAccessLayer(data_manager=None, middleware=[middleware], default_middleware=None)
+
+
+def test_dal_middleware_instantiation_not_needed():
+    """
+    Verify that you can specify method middleware as either an instance
+    or a class (as a convenience).
+    """
+    class SampleMiddleware:
+        def __call__(self, *args, **kwargs):
+            pass
+
+    dal = DataAccessLayer(data_manager=None, middleware=[SampleMiddleware], default_middleware=None)
+    assert isinstance(dal._reversed_middleware[0], SampleMiddleware)
```

Comment:
This would be better tested by making a call and expecting the middleware's result. The middleware would only return the result if it was instantiated properly. Then there would be no need to peek at internal variables.